### PR TITLE
Add an easter egg

### PR DESCRIPTION
**Pull request by Gymnasiast**
Figured OpenRCT2 needed an easter egg of its own.
(For those that don't get the reference, Google and Wikipedia will help.)

### CODE REVIEW

**Comment by IntelOrca** [comment]
Not sure about this...

**Comment by PFCKrutonium** [comment]
I'm okay with it - it doesn't unbalance the game, plus some day, someone will try it and be all like "whoa...:"

**Comment by rctdude2** [comment]
I'm fine with it as well.

**Comment by delebota** [comment]
Adding a bunch of money seems a little bit boring, but I'm not really sure what kind of behavior could be added either. I think it should at least be a clever name. Since you made it the digits of pi you could do Zhu Chongzhi who was the first person to correctly approximate pi's value to the number of digits you used.

Anyway, I think it would be cooler to use a contributer's name/handle and have a weird behavior like some of the ones that exist in the game already or maybe just a message from the contributers when you first change the name and nothing else.

**Comment by Enzemo** [comment]
I agree with delebota - it would make much more sense if it were associated with the names of the developers. Or maybe if just the names of the users in the github are added as random guest names, as a little easter egg. Most people don't notice the names unless they choose the peep anyway

**Comment by Gymnasiast** [comment]
I don't think those ideas are mutually exclusive. And I also agree we do need Easter eggs that take the names / handles of the developers. By the way, the Easter egg is a reference to Harry Enfield's Loadsamoney character, I just needed a big amount of money to go with it, and I found multiples of ten a bit boring.

**Comment by IntelOrca** [comment]
I just think, if one person adds one, every adds one and there will just be loads of PRs to add unnecessary trivial things to the game.

**Comment by Gymnasiast** [comment]
Hm, that could be a risk, indeed.

You probably know I try to keep the code base (at least relatively) clean, so I perfectly understand your objection to trivial or unnecessary things. But I have to say it isn't completely clear to me where you draw the line between useful and  unnecessary/trivial. The Explode! button was trivial in my eyes, the 'handymen mow by default' option completely unnecessary.

**Comment by Enzemo** [comment]
The 'handymen mow by default" is actually more a feature restoration from
RCT1. One of the project goals were to restore most the features that
weren't carried over - I'm certain that there are some players and some
parks where this feature would be invaluable.

As for the explode, I agree completely, unsure what it's purpose was
further than an aesthetic bit of fun

**Comment by medsouz** [comment]
I only added the "handymen mow by default" feature because I was already working in those files and saw it was planned on the wiki.

**Comment by Gymnasiast** [comment]
The handymen mow by default is hardly a feature, as it's completely useless. It was put on the wiki by someone, possibly in an attempt to get a complete list of changes, and IntelOrca put the entire list into a table.

The default was changed for a very good reason. Most players have no desire to have _any_ handymen mowing, and those that do might have a few, but certainly not most by default. I have said before that I'm against reintroducing every RCT1 quirk just for the sake of ticking a box. 

As the policy towards extra features and cheat seemed to be rather inclusive from IntelOrca's part - as in: almost all of them get accepted - I usually don't comment on usefulness, even though I don't like the pace at which the Options window and config files are currently growing. (By contrast, most cheats and Easter eggs require a lot less code and usually don't litter the Options window or config files.) I would probably be a bit more restrictive if I was project lead, actually. 

It seems inconsistent to me to include those features and not this one. I would be perfectly all right if this wouldn't get accepted, as I'm the first to admit it's trivial. But then I would like to see that applied to everything. After all, every feature - particularly those that require GUI elements and/or config settings - means more complex code, more configurations to support, a bigger codebase and a more cluttered interface (particularly the Options window).

**Comment by IntelOrca** [comment]
@Gymnasiast The mow by default is a preference and therefore is acceptable, but could be improved into a more temporary scheme where you can customise the items for new staff members from in the options window. It falls into the same category as setting the default inspection time for rides.

I agree that explode peeps is silly and not necessary and I wouldn't mind removing it. I think it was mostly done as a learning exercise.

You also already know I don't think we should be merging any changes or additions to actual gameplay logic right now unless it was very subtle and arguably an improvement and most likely unnoticeable such as peep behaviour.

So coming back to easter eggs, I think it would be better to at least have a discussion first with a large up vote and set of criteria for what would be allowed before going ahead.

When options window gets more cluttered, we will have to think about re-organising it by how likely a person is to changing it. E.g. resolution more likely than mow by default. Also perhaps "experimental features" like Chrome and other apps do. We can take OpenTTD as inspiration which have a search box on their advanced settings.

**Comment by Gymnasiast** [comment]
Personally I think OpenTTD actually has way too many settings. That includes a lot of settings that either nobody or an extremely small group is likely to change. I would argue that is the case for the 'mow by default' option, as there is no valid use case in my opinion. It could actually be made marginally useful by having checkboxes in the _staff window_ to change the default settings. That would also be a lot more flexible, as it would be easy to hire three handymen to mow grass, unchecking the box, and then hire three that don't.

I wouldn't be sad to see the explode peep feature go as well, although I hope I don't upset the author with it (LRFLEW). (Though he might be happy to know I would like to help him on the custom music feature instead.)

I do understand your issues with the interfering with gameplay. I figured my Easter egg wouldn't really do that, as it's confined to one guest, and renaming 2048 guests in different cases would take 3 hours if the player is quick enough to average 5 seconds per guest (which I consider highly unlikely). Not to mention there are far more efficient ways to gain money, even without cheating.

Regarding the Easter eggs discussion: I'm not that much a fan of popular vote, as it's only a way to decide, not to determine what's actually useful/good/fun/not interfering with gameplay etc.

Having said that though, if you look at the Easter Eggs name table, there are quite a lot of names in there. I would propose to keep the most well-known like the four Go Karts names, Chris Sawyer and Simon Foster, and replace the rest with some of our own. Having a separate discussion for that would be good as to decide who gets in, what the actions will be as not to interfere with gameplay, etc.

In conclusion, it might be best to close this one, get rid of the explode guests feature and rework the mow grass cheat into what I described above.

**Comment by IntelOrca** [comment]
I am not completely against easter eggs and maybe we can do this later on, but it seems a bit early at the moment, and in this case I did not know about the reference this easer eggs was making. People related to the original RCT and OpenRCT2 and memes related to RCT seem more appropriate.

**Comment by ThomasdenH** [comment]
I think OpenRCT should incorporate an intiutive settings window where a lot can be configured without being intrusive. OpenTTD does this, but not in an ideal way.

As for easter eggs and special things, I could see them work in an expansion pack of some sort, or a mod if you will. Of course such things may take a long time still.

**Comment by pizza2004** [comment]
Honestly, I know I don't really contribute much, although I'd love to, but I don't have any experience with decompiling code.  That said, I think that saying things like, "Explode is useless so it shouldn't exist." is extremely silly.  Any game designer puts little quirks into the game just for fun, and this discussion turned into a scary view of the future if we hate settings and weird little easter eggs.  I know that personally OpenTTD is fun /because/ of the settings.  I have to configure every setting the way I like it when I launch the game for maximum enjoyment, and I would be very surprised if most people don't at least go through the basic settings and change some of them.

I'm not looking to play RCT2, I'm looking to play a game that's built to play like it, but is infinitely more configurable and less strict.  Sure, that can detract from gameplay at times, such as winning being meaningless, but I'd argue that isn't as important since the original game had the balance.  What I'd love to see is just the ability to play RCT however I want, on whatever I want, and I really don't want it to turn into "This needs to be as slimline and focused on usefulness as possible." because a lot of the fun of this sort of open source project is that we get all the little options that don't really add or detract from gameplay, but just allow us to have it be new and different.

**Comment by Gymnasiast** [comment]
@pizza2004 : You haven't heard any of us saying it shouldn't exist. We said that we wouldn't be sad to it go, and that introducing new features - and therefore complexity in the code and the interface - requires a bit of thought and consideration. I haven't got a particular grudge against the feature, I gave it as an example.

We have accepted a lot of things that aren't necessarily useful, and we're not going down the "This needs to be as slimline and focused on usefulness as possible" route. Having said that though, my primary concern is that the code stays maintainable and the interface remains understandable. 

Some people like having a lot of options (IntelOrca and you, for example), but others get lost in the maze of options. And since OpenRCT2 is likely to get a _lot_ more options still, I'm saying that we should put more thought and consideration into what we put into the game, and how. I'm not saying we should cut everything unnecessary out. (After all, it's a game, not a piece of office software or something like that.)

If you look at all the PRs so far, you'll notice that I have voiced concerns a handful of times, and never actually objected inclusion. I'm not planning to act as a 'guard' either against everything I don't consider that useful. You don't have to be afraid we're pulling a GNOME.
